c10/core/DefaultDtype.cpp

```diff
@@ -2,15 +2,21 @@
 #include <c10/core/DefaultDtype.h>
 
 namespace c10 {
-static auto default_dtype = caffe2::TypeMeta::Make<float>();
-static auto default_complex_dtype = caffe2::TypeMeta::Make<c10::complex<float>>();
+static caffe2::TypeMeta default_dtype = caffe2::TypeMeta::Make<float>();
+static caffe2::TypeMeta default_complex_dtype = caffe2::TypeMeta::Make<c10::complex<float>>();
 
 void set_default_dtype(caffe2::TypeMeta dtype) {
-  default_dtype = std::move(dtype);
-  if(dtype == caffe2::TypeMeta::Make<double>()) {
-    default_complex_dtype = std::move(caffe2::TypeMeta::Make<c10::complex<double>>());
-  } else {
-    default_complex_dtype = std::move(caffe2::TypeMeta::Make<c10::complex<float>>());
+  default_dtype = dtype;
+  switch (default_dtype.toScalarType()) {
+    case ScalarType::Half:
+      default_complex_dtype = ScalarType::ComplexHalf;
+      break;
```

Comment:
I mean, there's nothing wrong with adding this case, but it's not going to do anything cuz ComplexHalf really doesn't work at all haha.